src/integrations/third-party/elementor.php

```diff
@@ -572,26 +532,44 @@ protected function get_metabox_post() {
 			return $this->post;
 		}
 
-		$post = null;
-		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Reason: We are not processing form information.
-		if ( isset( $_GET['post'] ) && \is_numeric( $_GET['post'] ) ) {
-			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized,WordPress.Security.NonceVerification.Recommended -- Reason: No sanitization needed because we cast to an integer,We are not processing form information.
-			$post = (int) \wp_unslash( $_GET['post'] );
-		}
+		$this->post = $this->request_post->get_post();
+
+		return $this->post;
+	}
 
-		if ( ! empty( $post ) ) {
-			$this->post = \get_post( $post );
+	/**
+	 * Retrieves the requested document ID from the POST request.
+	 *
+	 * Copied from the Elementor_Edit_Conditional class.
+	 *
+	 * @return int The requested document ID or 0 if not found.
+	 */
+	private function get_requested_document_id(): int {
```

Comment:
This seems mostly the same a the one in Request_Post only now it can return an 0 instead of null. Should these be reused somewhere or atleast both use the same return type?

Reply:
Oops, that is no longer used now that I moved it to the Request_Post!